guide/pooling-and-selective-consumer-service/tests/polling-and-selective-consumer_test.bal

```diff
@@ -0,0 +1,153 @@
+service<mb:Consumer> MaintenanceListener bind subscriberMaintenance {
+    onMessage(endpoint consumer, mb:Message message) {
+        http:Request outRequest;
+        //messageText3 = untaint check message.getTextMessageContent();
+        messageTexts[2] = untaint check message.getTextMessageContent();
+    }
+}
+service<mb:Consumer> ResearchListener bind subscriberResearch {
+    onMessage(endpoint consumer, mb:Message message) {
+        http:Request outRequest;
+        messageTexts[3] = untaint check message.getTextMessageContent();
+    }
+}
+function messagePublisher() {
+    json requestMessage1 = { "Message": "Earth Quake of 10 Richter Scale at the loaction near Panama", "AlarmStatus":
+    "ON", "DesasterRecoveryTeamStatus": "Dispatched" };
+    string msg1 = requestMessage1.toString();
+    mb:Message message1 = check queueSender.createTextMessage(msg1);
+    var p1 = message1.setPriority(1);
+    log:printInfo("P1 is:" + check message1.getPriority());
+    _ = queueSender->send(message1);
+    runtime:sleep(5000);
+    json requestMessage2 = { "AlarmStatus": "ON" };
+    string msg2 = requestMessage2.toString();
+    mb:Message message2 = check queueSender.createTextMessage(msg2);
+    var p2 = message2.setPriority(2);
+    log:printInfo("P2 is:" + check message2.getPriority());
+    _ = queueSender->send(message2);
+    runtime:sleep(5000);
+    json requestMessage3 = { "Maintenance": "Need to send a maintenance team to the sensor with SID 4338" };
+    string msg3 = requestMessage3.toString();
+    mb:Message message3 = check queueSender.createTextMessage(msg3);
+    var p3 = message3.setPriority(3);
+    log:printInfo("P3 is:" + check message3.getPriority());
+    _ = queueSender->send(message3);
+    runtime:sleep(5000);
+    json requestMessage4 = { "DATA": "lots of data need to analyze" };
+    string msg4 = requestMessage4.toString();
+    mb:Message message4 = check queueSender.createTextMessage(msg4);
+    var p4 = message4.setPriority(4);
+    log:printInfo("P4 is:" + check message4.getPriority());
+    _ = queueSender->send(message4);
+    runtime:sleep(5000);
+}
+@test:BeforeSuite
+function beforeFunc() {
+    // Start required services
+    _ = test:startServices("AuthorityListener");
+    _ = test:startServices("AlarmListner");
+    _ = test:startServices("MaintenanceListener");
+    _ = test:startServices("ResearchListener");
+    messagePublisher();
+}
+@test:AfterSuite
+function afterFunc() {
+    // Stop started services
+    _ = test:stopServices("AuthorityListener");
+    _ = test:stopServices("AlarmListner");
+    _ = test:stopServices("MaintenanceListener");
+    _ = test:stopServices("ResearchListener");
+}
+@test:Config
+function testNotifyAuthority() {
+    json requestMessage1 = { "Message": "Earth Quake of 10 Richter Scale at the loaction near Panama", "AlarmStatus":
+    "ON", "DesasterRecoveryTeamStatus": "Dispatched" };
+    string msg1 = requestMessage1.toString();
+    string expectedResponse = msg1;
```

Comment:
Shall we directly set expectedResponse = requestMessage1.toString(); without having 'string msg1' variable?

Reply:
Remove unwanted variables.